### PR TITLE
fix(html): mirror query clearing in URL

### LINKED ISSUE #33681: [Bug]: Playwright HTML report does not clear project filter from localhost URL, and reuses this previously used project filter when another filter is applied
### Version

1.49.0

### Steps to reproduce

1. Clone my repo at [https://github.com/apatelia/simple-pw-test](https://github.com/apatelia/simple-pw-test)
2. Install dependencies - `npm install`
3. Install browsers - `npx playwright install`
4. Run tests - `npx playwright test`
5. Open test report - `npx playwright show-report`
6. Notice that all four tests are displayed as passed under `example.spec.ts` file.
7. Click on `chromium` project badge from any test where it is displayed.
8. Notice that `p:chromium` is applied as a search filter in search box at the top of the report, and the URL is updated with `?q=p:chromium` querystring. Only two tests are displayed for `chromium` project.
9. Now, clear this `p:chromium` filter using `x` button that becomes available when mouse is hovered on the extreme right side of search box, as show below.
Image

10. Notice that search box is empty now and all four tests (chromium + Firefox) are displayed correctly. However, URL still contains the `?q=p:chromium` querystring.
11. Now, click on `link` badge displayed next to any of the `get started link` test.
12. Notice that `p:chromium` is reused along with `@link` as search filter even though it was not intended. Also, URL is updated with `?q=p:chromium%20@link` querystring. Also, only one `get started link` test is displayed instead of two tests (chromium + Firefox).
Image

### Expected behavior

I expect to see two (chromium + Firefox) `get started link` tests filtered and displayed instead of only one from `chromium` project, as I applied only `@link` filter and not `p:chromium` filter.

Image

### Actual behavior

It seems that `p:chromium` filter is not cleared from the URL and then it is reused along with `@link` filter, when only `@link` filter is applied.

### Additional context

This bug may be present in older versions as well, I noticed it only after upgrading to 1.49.

I am reporting this bug using Linux now, but I can reproduce this on Windows 11 as well.

### Environment

```shell
System:
    OS: Linux 6.11 Arch Linux
    CPU: (20) x64 12th Gen Intel(R) Core(TM) i7-12700K
    Memory: 22.18 GB / 31.06 GB
    Container: Yes
  Binaries:
    Node: 22.11.0 - ~/.nvm/versions/node/v22.11.0/bin/node
    Yarn: 1.22.22 - /usr/bin/yarn
    npm: 10.9.0 - ~/.nvm/versions/node/v22.11.0/bin/npm
    pnpm: 9.13.2 - ~/.local/share/pnpm/pnpm
  IDEs:
    VSCode: 1.95.3 - /usr/bin/code
  Languages:
    Bash: 5.2.37 - /usr/bin/bash
  npmPackages:
    @playwright/test: ^1.49.0 => 1.49.0
```

### PR DESCRIPTION
**Pull request by Skn0tt**
Closes https://github.com/microsoft/playwright/issues/33681

### CODE REVIEW

**Comment by pavelfeldman** [comment]
Current behavior is by design - look at what you've done to the history (try typing something long and then going back). I don't think this issue is work fixing - we don't want typing (and clearing) to affect the history. They are free to hit Enter to reset the url. Your semantics would not need the form btw, surprised you did non delete it!

File: packages/html-reporter/src/headerView.tsx [hunk 1]
```diff
@@ -49,13 +49,20 @@ export const HeaderView: React.FC<React.PropsWithChildren<{
...
       }>
         {icons.search()}
         {/* Use navigationId to reset defaultValue */}
         <input type='search' spellCheck={false} className='form-control subnav-search-input input-contrast width-full' value={filterText} onChange={e => {
-          setFilterText(e.target.value);
+          const filterText = e.target.value;
+          if (filterText)
+            setFilterText(filterText);
+          else
+            navigate('#');
```
Comment (pavelfeldman) [inline packages/html-reporter/src/headerView.tsx:65]: So if I'm mostly using keyboard and do Select all + Backspace or press and hold Backspace you want me to have a different history treatment? Not sure it is within the accepted UX guidelines. Why does Input have no "clearedUsingMouse" event for that use case?

**Comment by pavelfeldman** [comment]
I think "Submit", "Enter" or "Click" should navigate, while typing in the keyboard should not navigate. If input's clear button was accessible and had a dedicated handler, we could consider it a trigger event. But as it is now, clear button is just a shortcut for typing Backspace a bunch of times, or select all + backspace. That can't navigate.

**Comment by Skn0tt** [comment]
superceded by https://github.com/microsoft/playwright/pull/33754

**Pull request closed without merging**